core/src/main/scala/com/softwaremill/sttp/testing/SttpBackendStub.scala

```diff
@@ -99,12 +103,13 @@ class SttpBackendStub[R[_], S] private (
     def thenRespondWithCode(code: Int,
                             msg: String = ""): SttpBackendStub[R, S] = {
       val body = if (code >= 200 && code < 300) Right(msg) else Left(msg)
-      thenRespond(Response(body, code, msg, Nil, Nil))
+      thenRespondWithMonad(rm.unit(Response(body, code, msg, Nil, Nil)))
     }
     def thenRespond[T](body: T): SttpBackendStub[R, S] =
-      thenRespond(Response[T](Right(body), 200, "OK", Nil, Nil))
-    def thenRespond[T](resp: => Response[T]): SttpBackendStub[R, S] = {
-      val m: PartialFunction[Request[_, _], Response[_]] = {
+      thenRespondWithMonad(
+        rm.unit(Response[T](Right(body), 200, "OK", Nil, Nil)))
+    def thenRespondWithMonad(resp: => R[Response[_]]): SttpBackendStub[R, S] = {
```

Comment:
And maybe an overloaded version of `thenRespond(=> R[Response[_]])` would work as well?

Reply:
Unfortunately, this throws StackOverflowError, because after type erasure it matches `.thenRespond(body: T)`, which calls `.thanRespond` then, which matches `.thenRespond(body: T)`... 

We could change the name of method with `body: T` parameter, but I don't think that's a good idea ;)

Reply:
Ah right. Well, we'll have to live with that ;)